legistar/bills.py

```diff
@@ -414,17 +414,40 @@ def _version_rank(self, version):
 
     def relations(self, matter_id):
         relations = self.endpoint('/matters/{0}/relations', matter_id)
-        if relations:
-            highest_flag = max(int(relation['MatterRelationFlag'])
-                               for relation in relations)
 
-            relations = [relation for relation in relations
-                         if relation['MatterRelationFlag'] == highest_flag]
+        if relations:
+            return self._filter_relations(relations)
 
-            return relations
         else:
             return []
 
+    def _filter_relations(self, relations):
+        '''
+        Sometimes, many versions of a bill are related. This method returns the
+        most recent version of each relation. Override this method to apply a
+        different filter or return the full array of relations.
+        '''
+        # Sort relations such that the latest version of each matter
+        # ID is returned first.
+        sorted_relations = sorted(
+            relations,
+            key=lambda x: (
+                x['MatterRelationMatterId'],
+                x['MatterRelationFlag']
+            ),
+            reverse=True
+        )
+
+        # Assumes that there will not be more than 10 versions of a relation.
+        seen_relations = deque([], maxlen=10)
```

Comment:
why not just a set object here? seems like we don't really need to worry about memory ?